### PR TITLE
Add support for aggregate_metric_double field type

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

It adds support for the `aggregate_metric_double` field type.

## Why is it important?

This field type is used in TSDB use cases.

## Checklist

- [x] I have added test packages to [`test/packages`](https://github.com/elastic/package-spec/tree/main/test/packages) that prove my change is effective.
- [x] I have added an entry in [`spec/changelog.yml`](https://github.com/elastic/package-spec/blob/main/spec/changelog.yml).

## Related issues

- Part of https://github.com/elastic/package-spec/issues/457

### CODE REVIEW

**Review by kpollich — DISMISSED** [review]
LGTM 🚀

File: spec/integration/data_stream/fields/fields.spec.yml [hunk 1]
```diff
@@ -355,6 +367,46 @@ spec:
...
+      default_metric: true
+      metrics: true
+
+    allOf:
+      - if:
+          properties:
+            type:
+              const: aggregate_metric_double
+          required:
+            - type
```
Comment (jillguyonnet) [inline spec/integration/data_stream/fields/fields.spec.yml:380]: What does this add?
```
required:
  - type
 ```

File: spec/integration/data_stream/fields/fields.spec.yml [hunk 2]
```diff
@@ -355,6 +367,46 @@ spec:
...
+        then:
+          required:
+            - default_metric
+          properties:
+            default_metric:
```
Comment (jillguyonnet) [inline spec/integration/data_stream/fields/fields.spec.yml:385]: `type: string` is not needed here?

File: test/packages/bad_aggregate_metric_double/data_stream/foo/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+- name: no_metrics
+  type: aggregate_metric_double
+  default_metric: "max"
+- name: no_default_metrics
```
Comment (jillguyonnet) [inline test/packages/bad_aggregate_metric_double/data_stream/foo/fields/fields.yml:4]: Nit: typo
```suggestion
- name: no_default_metric
```

File: spec/integration/data_stream/fields/fields.spec.yml (hunk 2 above)
Comment (jsoriano) [reply spec/integration/data_stream/fields/fields.spec.yml:385]: `type: string` is defined in the referenced definition.

**Review by jillguyonnet — APPROVED** [review]
Nice!

**Pull request merged**